addons/web_editor/static/src/js/backend/field_html.js

```diff
@@ -32,22 +39,74 @@ var QWeb = core.qweb;
  *  - resizable
  *  - codeview
  */
-var FieldHtml = basic_fields.DebouncedField.extend(TranslatableFieldMixin, {
+var FieldHtml = basic_fields.DebouncedField.extend(TranslatableFieldMixin, DynamicPlaceholderFieldMixin, {
     description: _lt("Html"),
     className: 'oe_form_field oe_form_field_html',
     supportedFieldTypes: ['html'],
     isQuickEditable: true,
     quickEditExclusion: [
         '[href]',
     ],
-
     custom_events: {
         wysiwyg_focus: '_onWysiwygFocus',
         wysiwyg_blur: '_onWysiwygBlur',
         wysiwyg_change: '_onChange',
         wysiwyg_attachment: '_onAttachmentChange',
     },
 
+    /**
+     * Open a Model Field Selector which can select fields
+     * to create a dynamic placeholder <t-out> Element in the field HTML
+     * with or without a default text value.
+     *
+     * @override
+     * @public
+     * @param {String} baseModel
+     * @param {Array} chain
+     *
+     */
+    openDynamicPlaceholder: async function (baseModel, chain = []) {
+
+        let modelSelector;
+        const onFieldChanged = (ev) => {
+            this.wysiwyg.odooEditor.editable.focus();
+            if (ev.data.chain.length) {
+                let dynamicPlaceholder = "object." + ev.data.chain.join('.');
+                const defaultValue = ev.data.defaultValue;
+                dynamicPlaceholder += defaultValue && defaultValue !== '' ? ` or '''${defaultValue}'''` : '';
+
+                const t = document.createElement('T');
+                t.setAttribute('t-out', dynamicPlaceholder);
+                const fragment = new DocumentFragment();
+                fragment.appendChild(t);
+                this.wysiwyg.odooEditor.execCommand('insertFragment', fragment);
+                setSelection(...rightPos(t));
+                this.wysiwyg.odooEditor.editable.focus();
+            }
+            modelSelector.destroy();
+        };
+
+        const onFieldCancel = () => {
+            this.wysiwyg.odooEditor.editable.focus();
+            modelSelector.destroy();
+        }
+
+        modelSelector = await this._openNewModelSelector(
+            baseModel, chain, onFieldChanged, onFieldCancel
+        );
+
+        let top, left;
+        const position = getRangePosition(modelSelector.$el, this.wysiwyg.options.document);
+
+        top = position.top;
+        left = position.left - 25;
+        top = Math.min(top, window.window.innerHeight - modelSelector.$el.height() - 15);
+        left = Math.min(left, window.window.innerWidth - modelSelector.$el.width() - 10);
+
+        modelSelector.$el.css('top', top);
+        modelSelector.$el.css('left', left);
```

Comment:
The `_openNewModelSelector` method in the mixin already plays with css. Can't we have it right the first time rather than recomputing it differently here ?

Reply:
No we cannot, because the position here is base on the current wysiwyg selection. not the html field.